### DISCUSSION TITLE
Printing out debug messages

### DISCUSSION THREAD

**Discussion by ronsig** [redhat-developer/vscode-extension-tester#1322] [Q&A]
Sometimes I get an error that I'm having difficulty debugging, as it is timing related. The stack trace won't show me the point of failure in my test code.

E.g. 
![image](https://github.com/redhat-developer/vscode-extension-tester/assets/13620715/79165ff0-4598-4a28-b5bc-4db3ec2aa210)

Is there a way to log out debug message? Retry failed tests? Recover from this specific error by refreshing stale elements (e.g. WebDriver)?

Thanks

**Comment by djelinek** [comment]
Hello, this is I need to say a lack on `selenium-webdriver` side, many times this kind of errors can happen and it is hard to find out the right spot in source code.

Currently you can try to use change the logging level for webdriver by using extest binary parameter `-l`
```sh
-l, --log_level <level>                      # Log messages from webdriver with a given level (default: "Info")
```
you can pick one of possible options regarding to Selenium docs - JavaScript has 9 logger levels: OFF, SEVERE, WARNING, INFO, DEBUG, FINE, FINER, FINEST, ALL.

then the logs from webdriver can be found in `../test-resources/test.log` file which should be generated for each run of tests
